code/game/atoms_movable.dm

```diff
@@ -233,7 +233,8 @@
 		riding_datum.handle_vehicle_layer()
 		riding_datum.handle_vehicle_offsets()
 	for (var/datum/light_source/light as anything in light_sources) // Cycle through the light sources on this atom and tell them to update.
-		light.source_atom.update_light()
+		if(light.source_atom.light_system == STATIC_LIGHT)
+			light.source_atom.update_light()

```

Comment:
I don't think this will ever happen and you can probably leave it as it was to keep performance.

I'm pretty sure the overlay lights are not added to the parent's light sources.